tests/unit/dom/select-test.js

```diff
@@ -0,0 +1,210 @@
+  let context, element;
+
+  hooks.beforeEach(function () {
+    context = {};
+  });
+
+  hooks.afterEach(async function () {
+    if (element) {
+      element.setAttribute('data-skip-steps', true);
+      element.parentNode.removeChild(element);
+    }
+    if (context.owner) {
+      await teardownContext(context);
+    }
+
+    document.getElementById('ember-testing').innerHTML = '';
+  });
+
+  test('select without target', async function (assert) {
+    await setupContext(context);
+    assert.rejects(select(), /Must pass an element or selector to `select`./);
+  });
+
+  test('select without options', async function (assert) {
+    element = buildInstrumentedElement('div');
+
+    await setupContext(context);
+
+    assert.rejects(
+      select(element, undefined),
+      /Must provide an `option` or `options` to select when calling `select`./
+    );
+    assert.rejects(
+      select(element, null),
+      /Must provide an `option` or `options` to select when calling `select`./
+    );
+  });
+
+  test('select with unfindable selector ', async function (assert) {
+    element = buildInstrumentedElement('div');
+    const option = 'example';
+
+    await setupContext(context);
+
+    assert.rejects(
+      select('#fake-selector', option),
+      /Element not found when calling `select\('#fake-selector'\)`/
+    );
+  });
+
+  test('select with element that is not a HTMLSelectElement', async function (assert) {
+    element = buildInstrumentedElement('input');
+
+    const option = 'example';
+
+    await setupContext(context);
+
+    assert.rejects(
+      select(element, option),
+      `Element is not a HTMLSelectElement when calling \`select(${element})\``
+    );
+  });
+
+  test('select with element that is disabled', async function (assert) {
+    element = buildInstrumentedElement('select');
+    element.disabled = true;
+    const option = 'example';
+
+    await setupContext(context);
+
+    assert.rejects(
+      select(element, option),
+      `Element is disabled when calling \`select(${element})\`./`
+    );
+  });
+
+  test('select | multiple: false - options.length > 1', async function (assert) {
+    element = buildInstrumentedElement('select');
+    element.multiple = false;
+    const options = ['apple', 'orange'];
+
+    await setupContext(context);
+
+    assert.rejects(
+      select(element, options),
+      `HTMLSelectElement \`multiple\` attribute is set to \`false\` but multiple options have been passed when calling \`select(${element})\``
+    );
+  });
+
+  test('select | 4 options - multiple: true - optionsToSelect.length : 2', async function (assert) {
+    const optionValues = ['apple', 'orange', 'pineapple', 'pear'];
+    element = buildInstrumentedElement('select');
+    element.multiple = true;
+
+    optionValues.forEach(optionValue => {
+      const optionElement = buildInstrumentedElement('option');
+      optionElement.value = optionValue;
+      element.append(optionElement);
```

Comment:
Can you tweak this to `.appendChild` (`.append` doesn't work in IE11)?

```suggestion
      element.appendChild(optionElement);
```